cpp/serve/request.h

```diff
@@ -0,0 +1,222 @@
+/*!
+ *  Copyright (c) 2023 by Contributors
+ * \file request.h
+ * \brief Implementation of llm chat.
+ */
+#ifndef MLC_LLM_SERVE_REQUEST_H_
+#define MLC_LLM_SERVE_REQUEST_H_
+
+#include <tvm/node/reflection.h>
+#include <tvm/node/structural_equal.h>
+#include <tvm/node/structural_hash.h>
+#include <tvm/runtime/container/array.h>
+#include <tvm/runtime/container/optional.h>
+#include <tvm/runtime/container/shape_tuple.h>
+#include <tvm/runtime/container/string.h>
+#include <tvm/runtime/object.h>
+
+#include "config.h"
+#include "input.h"
+
+namespace mlc {
+namespace llm {
+namespace serve {
+
+using namespace tvm::runtime;
+
+/****************** Request ******************/
+
+/*!
+ * \brief The user submitted text-generation request, which contains
+ * a list of multi-modal inputs and a set of sampling configuration
```

Comment:
add note: request is immutable and can be re-dispatched to another node and restart the request handling